SpatialGDK/Source/SpatialGDKTests/Private/RPCContainer/RPCContainerTest.cpp

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "ObjectDummy.h"
+#include "ObjectSpy.h"
+
+#include "Utils/RPCContainer.h"
+#include "Schema/RPCPayload.h"
+
+#include <Core.h>
+
+#define RPCCONTAINER_TEST(TestName) \
+	TEST(FRPCContainer, TestName)
+
+using namespace SpatialGDK;
+
+namespace
+{
+	ESchemaComponentType AnySchemaComponentType = ESchemaComponentType::SCHEMA_ClientReliableRPC;
+	ESchemaComponentType AnyOtherSchemaComponentType = ESchemaComponentType::SCHEMA_ClientUnreliableRPC;
+
+	FUnrealObjectRef GenerateObjectRef(UObject* TargetObject)
+	{
+		return FUnrealObjectRef{ Worker_EntityId(TargetObject), 0 };
+	}
+
+	uint32 GeneratePayloadFunctionIndex()
+	{
+		static uint32 FreeIndex = 0;
+		return FreeIndex++;
+	}
+
+	FPendingRPCParamsPtr CreateMockParameters(UObject* TargetObject, ESchemaComponentType Type)
+	{
+		// Use PayloadData as a place to store RPC type
+		RPCPayload Payload(0, GeneratePayloadFunctionIndex(), SpyUtils::TypeToArray(Type));
+		int ReliableRPCIndex = 0;
+
+		FUnrealObjectRef ObjectRef = GenerateObjectRef(TargetObject);
+
+		return MakeUnique<FPendingRPCParams>(ObjectRef, MoveTemp(Payload), ReliableRPCIndex);
+	}
+}
+
+RPCCONTAINER_TEST(GIVEN_a_container_WHEN_nothing_has_been_added_THEN_nothing_is_queued)
+{
+	UObjectDummy* TargetObject = NewObject<UObjectDummy>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr Params = CreateMockParameters(TargetObject, AnySchemaComponentType);
+	FRPCContainer RPCs;
+
+	TestFalse("No queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, AnySchemaComponentType));
+
+    return true;
+}
+
+RPCCONTAINER_TEST(GIVEN_a_container_WHEN_one_value_has_been_added_THEN_it_is_queued)
+{
+	UObjectDummy* TargetObject = NewObject<UObjectDummy>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr Params = CreateMockParameters(TargetObject, AnySchemaComponentType);
+	FRPCContainer RPCs;
+
+	RPCs.QueueRPC(MoveTemp(Params), AnySchemaComponentType);
+
+	TestTrue("Queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, AnySchemaComponentType));
+
+    return true;
+}
+
+RPCCONTAINER_TEST(GIVEN_a_container_WHEN_multiple_values_of_same_type_have_been_added_THEN_they_are_queued)
```

Comment:
While I like the specificity it might be easier to make the name shorter so it can be found easily. Then have this as a comment.

Reply:
In this case MultipleValuesOfTypeAreQueued could do. This is the name that appears in the test runner so you don't need to know exactly what it is doing from it. You need to know roughly what it is and be able to read the name easily to then look it up.

Reply:
You will see this name in Automation Framework. So it is helpful when it's self-explanatory.

Reply:
It's not helpful if it's hard to read though. And importantly hard to then distinguish from the others.

Reply:
This long enough that it's going to end up being wrapped in most places you read it.

Reply:
It gets easy to read when you get used to it. I'm also against long variable names, but in case of tests - I would prefer to keep them follow the rules of GIVEN_WHEN_THEN and self explanatory.

Reply:
I think writing given-when-then is fine somewhere in the test but there is nothing you can do with that information in the test runner, you have to read the test to work out what has actually gone wrong and these names hinder that.

Reply:
This is much harder to read than a shorter name directing you to which test needs to be looked at.

Reply:
The test itself should do exactly what the name states.

Reply:
That can still be the case. But also that seems like a softer rule.

Reply:
The actual workflow is you write tests that do something obvious and can be easily read. Later you make changes, and see something has broken. You then look at the test in question to know exactly what has broken.

The name will never tell you the entirety of a test so you will normally have to read the test unless something really obvious went wrong. So the thing the name needs to give you is some information about what the test is and where to look to read the actual test. A wall of text and names that require significant effort to read do not accomplish that and those comments could be more clearly written as comments.

Reply:
(FYI I won't review with request-changes for a naming dispute and this might be easier to argue about offline)

Reply:
Discussed offline. We will use this convention.